siac/rentercmd.go

```diff
@@ -110,21 +130,107 @@ func abs(path string) string {
 	return abspath
 }
 
-func renterdownloadqueuecmd() {
+func filterFileInfos(fis []modules.FileInfo, f func(modules.FileInfo) bool) []modules.FileInfo {
+	fisFiltered := make([]modules.FileInfo, 0)
+	for _, fi := range fis {
+		if f(fi) {
+			fisFiltered = append(fisFiltered, fi)
+		}
+	}
+	return fisFiltered
+}
+
+func filterDownloadInfos(dis []modules.DownloadInfo, f func(modules.DownloadInfo) bool) []modules.DownloadInfo {
+	disFiltered := make([]modules.DownloadInfo, 0)
+	for _, di := range dis {
+		if f(di) {
+			disFiltered = append(disFiltered, di)
+		}
+	}
+	return disFiltered
+}
```

Comment:
filtering functions are probably overkill unless we expect to use them for multiple commands.
shed a tear for Go's lack of generics, then move on.

also, for future reference, prefer `var slice []T` to `slice := make([]T, 0)` (unless it's going to be encoded as JSON)